ballerina/document-types.bal

```diff
@@ -108,3 +108,12 @@ public type FileDocument record {|
     # File content, a URL, binary data, or a file ID reference
     byte[]|Url|FileId content;
 |};
+
+# Represents a binary document.
+public type BinaryDocument record {|
```

Comment:
Doesn't this conflict with the other document kinds (that also allow binary data)?

Reply:
I meant more at a conceptual level. It gets a bit complicated with mutable values so might not be a direct subtype expectation, but for example, an (immutable) value of type `ai:ImageDocument` with a `byte[]` value set to `content` won't be a subtype of this `ai:BinaryDocument` type, which feels kinda wrong.

Reply:
@SasinduDilshara we can still do this right?